src/charm.py

```diff
@@ -768,6 +815,9 @@ def _install_deps(self) -> None:
                 "security.port_isolation=true",
             ]
         )
+        # install aproxy, currently only edge channel available
+        execute_command(["snap", "install", "aproxy", "--edge"])
```

Comment:
Wouldn't it be better to set up the aproxy within the runner machine, rather than within the charm unit? That would allow users to easily access and review the aproxy logs inside the GitHub workflows to identify any proxy issues, such as rejections caused by upstream proxy server ACL rules.

Reply:
Have updated the code accordingly